src/vivarium/examples/boids/location_swarming.py

```diff
@@ -0,0 +1,176 @@
+from typing import Any, Dict, List
+
+import numpy as np
+import pandas as pd
+
+from vivarium import Component
+from vivarium.framework.engine import Builder
+from vivarium.framework.population import SimulantData
+
+
+class Location(Component):
+    ##############
+    # Properties #
+    ##############
+    @property
+    def configuration_defaults(self) -> Dict[str, Any]:
+        return {
+            "location": {
+                "width": 1000,  # Width of our field
+                "height": 1000,  # Height of our field
+                "max_velocity": 2,
+                "separation_distance": 30,
+                "separation_force": 0.03,
+                "cohesion_force": 0.03,
+                "alignment_force": 0.03,
+            },
+        }
+
+    @property
+    def columns_created(self) -> List[str]:
+        return ["x", "vx", "y", "vy"]
+
+    #####################
+    # Lifecycle methods #
+    #####################
+
+    def setup(self, builder: Builder) -> None:
+        self.config = builder.configuration.location
+
+        self.neighbors = builder.value.get_value("neighbors")
+
+    ########################
+    # Event-driven methods #
+    ########################
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        count = len(pop_data.index)
+        # Start randomly distributed, with random velocities
+        new_population = pd.DataFrame(
+            {
+                "x": self.config.width * np.random.random(count),
+                "y": self.config.height * np.random.random(count),
+                "vx": ((2 * np.random.random(count)) - 1) * self.config.max_velocity,
+                "vy": ((2 * np.random.random(count)) - 1) * self.config.max_velocity,
+            },
+            index=pop_data.index,
+        )
+        self.population_view.update(new_population)
+
+    def on_time_step(self, event):
```

Comment:
I think we need to break this method down with helper methods to improve readability.

Reply:
Yep makes sense. See my question about separate components above.